pallets/connectors/src/lib.rs

```diff
@@ -706,6 +706,25 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::weight(10_000)]
+		#[pallet::call_index(10)]
+		pub fn schedule_upgrade(origin: OriginFor<T>, contract: DomainAddress) -> DispatchResult {
+			match contract {
+				DomainAddress::EVM(_, _) => {
+					let who = ensure_signed(origin)?;
+
+					T::OutboundQueue::submit(
+						who,
+						contract.domain(),
+						Message::ScheduleUpgrade {
+							contract: contract.address(),
+						},
+					)
+				}
+				_ => Err(DispatchError::Other("Not an EVM domain address")),
+			}
```

Comment:
This is not wrong but a more idiomatic way to implement this is to have an `ensures!(` block where we would ensure the `contract` domain is not `Centrifuge` and fail otherwise with `Error::<T>::InvalidDomain`. Something like:

```suggestion
			let who = ensure_signed(origin)?;
			ensure!(contract.domain() != Domain::Centrifuge, Error::<T>::InvalidDomain);
      
			T::OutboundQueue::submit(
			  who,
			  contract.domain(),
			  Message::ScheduleUpgrade {
			    contract: contract.address(),
			  },
			)
```